### PR TITLE
Add plan-based monthly quota tracking (#39)

### LINKED ISSUE #39: Plan-based monthly quota tracking (complement to #5 block forecasting)
## Context

#5 covers the 5-hour rolling window forecasting. This issue covers the **other axis** of quota tracking: monthly plan budgets. Different time scale, different concern.

A user on Claude Max at \$200/mo cares about monthly runway more than the next 5-hour reset. A user on Cursor Pro at \$20/mo wants to know when they'll hit the wall before the next billing cycle. These are orthogonal questions to the 5-hour-window forecast.

Pattern from codeburn's `src/plan-usage.ts:1-149` and `src/plans.ts:1-56`.

## Scope

Three preset plans out of the box:

- **Claude Pro** — \$20/mo, usage metered against plan allotment
- **Claude Max** — \$200/mo
- **Cursor Pro** — \$20/mo

Extensible via user config file (`\$RELAYBURN_HOME/plans.json`) for other providers / tiers.

## Per-plan state

```ts
interface Plan {
  id: string;
  provider: 'claude' | 'cursor' | 'custom';
  name: string;
  budgetUsd: number;
  resetDay: number;        // 1-31 for day of month; or 'calendar' for month boundary
}

interface PlanUsage {
  plan: Plan;
  spentUsd: number;
  daysElapsed: number;
  daysInCycle: number;
  projectedEndOfCycleUsd: number;  // linear extrapolation from current rate
  overBudget: boolean;
  runwayDays: number | null;       // null if under current pace; integer days otherwise
  resetAt: string;                 // ISO timestamp
}
```

## CLI

```
burn plans                                # list configured plans + current status
burn plans add --provider claude --plan pro
burn plans set-reset-day 15               # custom reset cycle (not calendar month)
burn plans remove <id>

burn limits                                # includes plan status alongside 5-hour window
```

Integrate into `burn limits` output:

```
Claude
  5-hour window:
    OAuth:     34% used     resets in 2h 14m
    Forecast:  87% projected at block-end

  Monthly plan (Claude Max):
    Spent:     $87.42 / $200.00   (43%)
    Elapsed:   13 / 30 days
    Projected: $201.73 end-of-cycle (0.9% over)
    Runway:    29 more days at current rate
```

## Key difference from #5

- #5 (block forecasting) — derived from local ledger data, no network. Projects into the next few hours.
- This issue — derived from local ledger *plus* user-configured plan budgets. Projects into the next weeks.

They compose: `burn limits` shows both. A user who's within their 5-hour window but on pace to exceed their monthly plan sees different advice than the reverse.

## Custom reset cycles

Claude Max's billing cycle isn't necessarily the calendar month. Users can configure `resetDay`:

- `1` — calendar month (default)
- `15` — reset on the 15th
- `'anniversary'` — reset on the day-of-month they signed up

Codeburn supports this in `plan-usage.ts`; we can mirror the same semantics.

## Acceptance

- `burn plans add --provider claude --plan max` registers a plan with known defaults.
- `burn limits` shows monthly status alongside 5-hour status when a plan is configured.
- Projection is honest about confidence: less than 7 days of data in a cycle → mark projection as `(limited data)`.
- Over-budget case shows clearly: *\"$201 projected on $200 plan; $1 over\"*.
- Custom `resetDay` cycles are respected.

## Depends on

- #4 (ledger dedup + cursor) — plans query the ledger for spend; needs efficient querying.
- #5 (block forecasting) — so `burn limits` can surface both axes together.

## Priority

**Medium.** Complements #5 rather than extending it. Users on Pro/Max plans will want this; users on pay-per-token API access won't care. File but don't prioritize before #37/#38.

### PR DESCRIPTION
**Pull request by willwashburn**
## Summary

Closes #39 — complement to `burn limits`'s 5-hour OAuth window. Tracks spend against monthly plan budgets (Claude Pro/Max, Cursor Pro, custom) and surfaces projected end-of-cycle spend, runway-days-at-current-rate, and over/under-budget delta. Plans are configured via a new `burn plans` subcommand and persisted to `~/.relayburn/plans.json`; status flows automatically into `burn limits` output as a `Monthly plan` block alongside the 5-hour view.

## What ships

**`@relayburn/ledger`** — config primitives:
- `Plan` / `PlansFile` / `PlanProvider` types, `loadPlans()` / `savePlans()` against `~/.relayburn/plans.json` (override via `RELAYBURN_HOME`).
- `BUILTIN_PRESETS` covering claude/pro ($20), claude/max ($200), cursor/pro ($20). `findPreset(provider, name)` looks them up.
- `normalizePlan` validates each row (positive `budgetUsd`, integer `resetDay` 1-31, known `provider`) so a malformed `plans.json` throws once at load with a path-prefixed error rather than producing garbage downstream.

**`@relayburn/analyze`** — math:
- `computePlanUsage(plan, turns, { pricing, now })` aggregates spend over the active cycle and returns `PlanUsage`: `spentUsd`, `daysElapsed`, `daysInCycle`, `projectedEndOfCycleUsd` (linear extrapolation), `overBudget`, `runwayDays` (only populated when over-pace — runway is "until the wall", not "until the cycle ends"), `resetAt`, `limitedData` (true when fewer than 7 days have elapsed).
- `cycleBounds(resetDay, now)` — UTC-anchored, end-of-month clamping for `resetDay > 28` (e.g. resetDay=31 in February resolves to Feb 28/29), year-boundary safe, never returns a zero-length cycle.
- Provider-aware filtering: `claude` plans count `claude-code` + `anthropic-api` turns; `cursor` plans count `cursor` turns (no reader emits these yet — the branch is in place for when a Cursor adapter lands); `custom` plans count every turn.

**`@relayburn/cli`** — surface:
- `burn plans` subcommand: `add --provider <p> --preset <name>` for built-ins; `add --provider custom --id <id> --name "<…>" --budget <usd> [--reset-day <1-31>]` for custom. `remove <id>`, `set-reset-day <id> <day>`. Bare `burn plans` lists configured plans with cycle spend, projection, budget, and elapsed days. `--json` on the list view.
- `burn limits` integrates plan status: a `Monthly plan (<name>):` block per plan reports `Spent / Budget`, `Elapsed`, `Projected end-of-cycle ($X over | Y% under)`, and `Runway: N more days at current rate` when over-pace. Projections with `limitedData` render with a `(limited data)` marker per the issue's acceptance. The `--json` payload gains a `plans[]` array with the full numeric breakdown.

## Acceptance criteria check

- [x] `burn plans add --provider claude --plan max` registers a plan with known defaults — works with `--preset max` (issue spec said `--plan` but the implementation uses `--preset` to disambiguate from the verb sense; documented in `burn plans help`).
- [x] `burn limits` shows monthly status alongside 5-hour status when a plan is configured.
- [x] Projection is honest about confidence: `< 7 days` of cycle data → `(limited data)` marker.
- [x] Over-budget case shows clearly: `Projected: $201.73 end-of-cycle ($1.73 over)`.
- [x] Custom `resetDay` cycles respected (anniversary-style billing).

## Test plan

- [x] `pnpm run test:ts` passes — **314 tests** (+36 new across `plans` config / `plan-usage` math / `burn plans` CLI / `burn limits` integration).
- [x] `cycleBounds` covered for: calendar (day 1), mid-cycle anniversary (day 15), pre-anniversary (day 15 from day 10), leap-year clamping (day 31 in Feb), end-of-month roll-forward, year-boundary crossing.
- [x] `computePlanUsage` covered for: cycle filtering, linear projection math, overBudget + runway, runway-null-when-under-budget, limitedData flag, zero-projection-without-NaN at cycle start, claude-vs-custom provider filtering, anniversary cycle, garbage timestamp tolerance.
- [x] `burn plans` CLI covered for: empty list message, preset add, `--reset-day` override on preset, invalid `--reset-day`, missing custom flags, custom add, duplicate id refusal, remove, remove unknown id, `set-reset-day` happy + invalid path, unknown subcommand.
- [x] `burn limits` covered for: Monthly plan block rendering, `(limited data)` marker, `plans[]` in JSON output.
- [x] End-to-end smoke against my real ledger — `burn plans add --provider claude --preset max` then `burn plans` showed actual cycle spend computed from claude-code session logs.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/agentworkforce/burn/pull/66" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Review by barryollama — APPROVED** [review]
## Review Summary

Excellent work on implementing plan-based monthly quota tracking for #39.

**What's good:**
- Clean separation across , , and  packages
- Robust cycle math with proper handling for month-end clamping (day 31 → 28/29 in Feb), year boundaries, and leap years
- Comprehensive test coverage including edge cases for  warnings, projection calculations, and provider filtering
- Built-in presets (Claude Pro/Max, Cursor Pro) provide good UX out of the box
- The  flag on projections with <7 days is a nice touch to prevent users anchoring on early-cycle noise

**Nit (non-blocking):**
- The  provider branch in  currently returns false always (no readers emit this source yet). Consider adding a reference to the planned Cursor adapter issue/PR when it lands.

LGTM! ✅

**Review by barryollama — APPROVED** [review]
LGTM. Clean architecture across the three packages with comprehensive test coverage. The cycle bounds logic handles edge cases well (month-end clamping, leap years). Built-in presets provide good UX. The (limited data) flag for early-cycle projections is a thoughtful addition to prevent anchoring on first-week noise. The cursor provider branch is documented as awaiting future adapter.

**Comment by willwashburn** [comment]
Thanks for the review! On the cursor-branch nit — sharper than I had it: per #22 (closed wontfix), Cursor moved usage tracking server-side in early 2026 and the local SQLite token fields zeroed out, so there isn't a future adapter coming. Updated in 9f311ad:

- `plan-usage.ts:matchesProvider` — replaced the "once a Cursor adapter lands" comment with the actual reason + `#22` reference so future readers don't re-investigate.
- `plans.ts` CLI help — annotated the cursor preset with "spend tracking unavailable (see #22)" so users see the limitation before adding the plan.
- `plans.ts` add path — prints a one-line note after `burn plans add --provider cursor --preset pro` explaining that spend will report $0. The preset stays in (it matches the issue spec, and the cursor branch is still useful as a budget/reset-day registry) but the limitation is now surfaced rather than buried.

File: packages/cli/src/commands/limits.ts [hunk 1]
```diff
@@ -95,6 +100,8 @@ export async function runLimits(args: ParsedArgs, deps: LimitsDeps = {}): Promis
       ? projectFromOauth(usage.five_hour.percent_used, forecast.data)
       : null;
 
+    const planStatuses = await loadPlanStatuses();
```
[the remark this answers is not in the document]
Comment (willwashburn) [reply packages/cli/src/commands/limits.ts]: Real bug — fixed in 3c1868f. Wrapped `loadPlanStatuses()` in try/catch matching the existing `fetchOnce` pattern: a malformed `plans.json` now warns on stderr (`[burn] warning: could not load plans (...)`) and degrades to an empty plan list. The 5-hour quota block, forecast, and `--watch` loop all keep working. Added a regression test injecting a throwing `loadPlanStatuses` and asserting exit 0 + the warning lands on stderr + the OAuth render proceeds without a Monthly plan block.
